packages/astro/src/actions/runtime/virtual/server.ts

```diff
@@ -212,3 +230,80 @@ function unwrapBaseObjectSchema(schema: z.ZodType, unparsedInput: FormData) {
 	}
 	return schema;
 }
+
+/**
+ * Access information about Action requests from middleware.
+ */
+export function getMiddlewareContext(context: APIContext) {
+	const callerInfo = getCallerInfo(context);
+
+	// Prevents action results form being handled on a rewrite.
+	// Also prevents our *own* fallback middleware from running
+	// if the user's middleware has already handled the result.
+	const actionResultAlreadySet = Boolean((context.locals as Locals)._actionPayload);
+
+	let action:
+		| {
+				/** Whether an action was called using an RPC function or by using an HTML form action. */
+				calledFrom: 'rpc' | 'form-action';
+				/** The name of the action. Useful to track the source of an action result during a redirect. */
+				name: string;
+				/** Programatically call the action to get the result. */
+				handler: () => Promise<SafeResult<any, any>>;
+		  }
+		| undefined = undefined;
+
+	if (callerInfo && context.request.method === 'POST' && !actionResultAlreadySet) {
```

Comment:
What about `DELETE` and `PATCH` methods? Or maybe actions work only with `POST` methods?

Reply:
Right, we just use POST for RPCs. Since RPC functions don't implement REST naming conventions, they'll typically be POST only. Follows React server actions and tRPC